### PR TITLE
Refactor the flag test of BBF_PROF_WEIGHT

### PR DESCRIPTION
**Pull request by briansull**
We now always call the new method hasProfileWeight() to perform the flag test.

### CODE REVIEW

**Comment by briansull** [comment]
@dotnet/jit-contrib PTAL

File: src/jit/block.h [hunk 1]
```diff
@@ -521,16 +521,31 @@ struct BasicBlock : private LIR::Range
     // getBBWeight -- get the normalized weight of this block
     unsigned getBBWeight(Compiler* comp);
 
-    // setBBWeight -- if the block weight is not derived from a profile, then set the weight to the input
-    // weight, but make sure to not overflow BB_MAX_WEIGHT
+    // hasProfileWeight -- Returns true if this block's weight came from profile data
+    bool hasProfileWeight()
```
Comment (pgavlin) [inline src/jit/block.h]: Nit: this function should be marked `const`.

**Comment by AndyAyersMS** [comment]
This should be no-diff, right...?

File: src/jit/block.h (hunk 1 above)
Comment (briansull) [inline src/jit/block.h]: OK

**Comment by briansull** [comment]
Right there are no asm diffs

**Comment by briansull** [comment]
@dpodder /cc

File: src/jit/optimizer.cpp [hunk 2]
```diff
@@ -370,7 +370,7 @@ void Compiler::optUnmarkLoopBlocks(BasicBlock* begBlk, BasicBlock* endBlk)
             // Don't unmark blocks that are set to BB_MAX_WEIGHT
             // Don't unmark blocks when we are using profile weights
             //
-            if (!curBlk->isMaxBBWeight() && ((curBlk->bbFlags & BBF_PROF_WEIGHT) == 0))
+            if (!curBlk->isMaxBBWeight() && curBlk->hasProfileWeight())
```
Comment (dpodder) [inline src/jit/optimizer.cpp]: Shouldn't this be negated: `!curBlk->hasProfileWeight()`?

Comment (briansull) [inline src/jit/optimizer.cpp]: Yes, Thanks for catching that

**Comment by briansull** [comment]
@dotnet-bot retest CentOS7.1 x64 Debug Build and Test
@dotnet-bot retest FreeBSD x64 Checked Build
@dotnet-bot retest Linux ARM Emulator Cross Debug Build
@dotnet-bot retest Linux ARM Emulator Cross Release Build
@dotnet-bot retest OSX10.12 x64 Checked Build and Test
@dotnet-bot retest Ubuntu x64 Checked Build and Test
@dotnet-bot retest Ubuntu x64 Formatting
@dotnet-bot retest Windows_NT arm Cross Debug Build
@dotnet-bot retest Windows_NT arm Cross Release Build
@dotnet-bot retest Windows_NT arm64 Cross Debug Build
@dotnet-bot retest Windows_NT x64 Debug Build and Test
@dotnet-bot retest Windows_NT x64 Formatting
@dotnet-bot retest Windows_NT x64 Release Priority 1 Build and Test
@dotnet-bot retest Windows_NT x86 Checked Build and Test

**Comment by briansull** [comment]
@dpodder I made the requested change

**Comment by dpodder** [comment]
@briansull LGTM, thanks!

**Comment by briansull** [comment]
@dotnet-bot retest OSX10.12 x64 Checked Build and Test
@dotnet-bot retest Ubuntu x64 Checked Build and Test
@dotnet-bot retest Ubuntu x64 Formatting
@dotnet-bot retest Windows_NT arm Cross Debug Build
@dotnet-bot retest Windows_NT arm Cross Release Build
@dotnet-bot retest Windows_NT arm64 Cross Debug Build
@dotnet-bot retest Windows_NT x64 Debug Build and Test
@dotnet-bot retest Windows_NT x64 Formatting
@dotnet-bot retest Windows_NT x64 Release Priority 1 Build and Test
@dotnet-bot retest Windows_NT x86 Checked Build and Test

**Comment by briansull** [comment]
@dotnet-bot retest Windows_NT x64 Debug Build and Test
@dotnet-bot retest Windows_NT x64 Formatting
@dotnet-bot retest Windows_NT x64 Release Priority 1 Build and Test
@dotnet-bot retest Windows_NT x86 Checked Build and Test

**Pull request merged**